zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -291,6 +410,36 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
   final def race[R1 <: R, E1 >: E, A1 <: A, B1 >: B](other: Http[R1, E1, A1, B1]): Http[R1, E1, A1, B1] =
     Http.Race(self, other)
 
+  /**
+   * Keeps some of the errors, and terminates the http app with the rest.
+   */
+  final def refineOrDie[E1](
+    pf: PartialFunction[E, E1],
+  )(implicit ev1: E <:< Throwable, ev2: CanFail[E]): Http[R, E1, A, B] =
+    refineOrDieWith(pf)(ev1)
+
+  /**
+   * Keeps some of the errors, and terminates the http app with the rest, using
+   * the specified function to convert the `E` into a `Throwable`.
+   */
+  final def refineOrDieWith[E1](pf: PartialFunction[E, E1])(f: E => Throwable)(implicit
+    ev: CanFail[E],
+  ): Http[R, E1, A, B] =
+    self.catchAll(err => (pf lift err).fold[Http[R, E1, A, B]](Http.die(f(err)))(Http.fail))
+
+  /**
+   * Returns an effect that semantically runs the http app with request value of
+   * type `A`, producing an [[zhttp.http.HExit]] for the completion value of the
+   * app.
+   */
+  final def run: Http[R, Nothing, A, HExit[R, E, B]] =
```

Comment:
```suggestion
  final def exit: Http[R, Nothing, A, HExit[Any, E, B]] =
```
- Like `either` we can keep it as `exit`. 
- I think `R` will be `Any` in `HExit`.

I wonder if this is a useful operator. Can you provide some examples for it?

Reply:
Well I use similar combinator at my work to implement `Http.zipPar` with accumulation of failures, but it's a niche application so indeed we can wait for more convincing examples from the community before introducing the combinator. I'm also not sure it's helpful right now.

Reply:
So I've removed questionable combinator and merged `main`, the PR is ready for continuation of review.